log/logger.go

```diff
@@ -15,56 +15,36 @@ var (
 
 func newLogger(c *Config) Logger {
 	return Logger{
-		Timestamp:         c.AdvancedSettings.TimeStampFormat,
-		Spacer:            c.AdvancedSettings.Spacer,
-		ErrorHeader:       c.AdvancedSettings.Headers.Error,
-		InfoHeader:        c.AdvancedSettings.Headers.Info,
-		WarnHeader:        c.AdvancedSettings.Headers.Warn,
-		DebugHeader:       c.AdvancedSettings.Headers.Debug,
-		ShowLogSystemName: *c.AdvancedSettings.ShowLogSystemName,
-	}
-}
-
-func (l *Logger) newLogEvent(data, header, slName string, w io.Writer) error {
-	if w == nil {
-		return errors.New("io.Writer not set")
-	}
-
-	pool, ok := eventPool.Get().(*[]byte)
-	if !ok {
-		return errors.New("unable to type assert slice of bytes pointer")
+		TimestampFormat:               c.AdvancedSettings.TimeStampFormat,
+		Spacer:                        c.AdvancedSettings.Spacer,
+		ErrorHeader:                   c.AdvancedSettings.Headers.Error,
+		InfoHeader:                    c.AdvancedSettings.Headers.Info,
+		WarnHeader:                    c.AdvancedSettings.Headers.Warn,
+		DebugHeader:                   c.AdvancedSettings.Headers.Debug,
+		ShowLogSystemName:             c.AdvancedSettings.ShowLogSystemName != nil && *c.AdvancedSettings.ShowLogSystemName,
+		BypassJobChannelFilledWarning: c.AdvancedSettings.BypassJobChannelFilledWarning,
 	}
-
-	*pool = append(*pool, header...)
-	if l.ShowLogSystemName {
-		*pool = append(*pool, l.Spacer...)
-		*pool = append(*pool, slName...)
-	}
-	*pool = append(*pool, l.Spacer...)
-	if l.Timestamp != "" {
-		*pool = time.Now().AppendFormat(*pool, l.Timestamp)
-	}
-	*pool = append(*pool, l.Spacer...)
-	*pool = append(*pool, data...)
-	if data == "" || data[len(data)-1] != '\n' {
-		*pool = append(*pool, '\n')
-	}
-	_, err := w.Write(*pool)
-	*pool = (*pool)[:0]
-	eventPool.Put(pool)
-
-	return err
 }
 
 // CloseLogger is called on shutdown of application
 func CloseLogger() error {
-	return GlobalLogFile.Close()
+	mu.Lock()
```

Comment:
I'm thinking that there is some fun stuff going on with closing the application
```
panic: send on closed channel

goroutine 933 [running]:
github.com/thrasher-corp/gocryptotrader/log.(*multiWriterHolder).StageLogEvent(0xc000652690, 0xc004e707e0, {0xc000417ee0, 0x10}, {0x171e550, 0x9}, {0xc000417ebc, 0x3}, {0xc000131b00, 0x15}, ...)
        C:/Users/ScottGrant/go/src/github.com/thrasher-corp/gocryptotrader/log/logger_multiwriter.go:113 +0x2a6
github.com/thrasher-corp/gocryptotrader/log.Debugf(0xc0129fb2c0?, {0x175641b, 0x2d}, {0xc006dc56e0, 0x2, 0x2})
        C:/Users/ScottGrant/go/src/github.com/thrasher-corp/gocryptotrader/log/loggers.go:105 +0x137
github.com/thrasher-corp/gocryptotrader/exchanges/stream.(*WebsocketConnection).ReadMessage(0xc00350a080)
        C:/Users/ScottGrant/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/stream/websocket_connection.go:243 +0x336
github.com/thrasher-corp/gocryptotrader/exchanges/kraken.(*Kraken).wsFunnelConnectionData(0x30003100330031?, {0x1924470, 0xc00350a080}, 0x7000730022003a?)
        C:/Users/ScottGrant/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/kraken/kraken_websocket.go:146 +0xa4
created by github.com/thrasher-corp/gocryptotrader/exchanges/kraken.(*Kraken).WsConnect
        C:/Users/ScottGrant/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/kraken/kraken_websocket.go:100 +0x1ec
exit status 2
```

```
panic: send on closed channel

goroutine 953 [running]:
github.com/thrasher-corp/gocryptotrader/log.(*multiWriterHolder).StageLogEvent(0xc0006c9380, 0xc00287d440, {0xc0001cd6e0, 0x10}, {0x144e550, 0x9}, {0xc0001cd6bc, 0x3}, {0xc0000bfbc0, 0x15}, ...)
        C:/Users/ScottGrant/go/src/github.com/thrasher-corp/gocryptotrader/log/logger_multiwriter.go:113 +0x2a6
github.com/thrasher-corp/gocryptotrader/log.Debugf(0xc000b84500?, {0x148641b, 0x2d}, {0xc007239220, 0x2, 0x2})
        C:/Users/ScottGrant/go/src/github.com/thrasher-corp/gocryptotrader/log/loggers.go:105 +0x137
github.com/thrasher-corp/gocryptotrader/exchanges/stream.(*WebsocketConnection).ReadMessage(0xc0005a2080)
        C:/Users/ScottGrant/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/stream/websocket_connection.go:243 +0x336
github.com/thrasher-corp/gocryptotrader/exchanges/bitfinex.(*Bitfinex).wsReadData(0x6d00000022?, {0x1654470, 0xc0005a2080})
        C:/Users/ScottGrant/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/bitfinex/bitfinex_websocket.go:88 +0xa2
created by github.com/thrasher-corp/gocryptotrader/exchanges/bitfinex.(*Bitfinex).WsConnect
        C:/Users/ScottGrant/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/bitfinex/bitfinex_websocket.go:56 +0x1f5
```

It looks like things still wish to send some logs in a racey fashion. You may wish to attempt to close the channel only after all logs are processed, but dismiss any new ones from being added (or potentially just fmt.Println them). An atomic may help with preventing new additions but I'll let you investigate further.

For reference, I'm running with the following `go run . --enableallexchanges --enableallpairs --exchangeverbose`
So there are many many things wanting to log


update: I believe `StageLogEvent` will be the culprit with the following:
```
	default:
		// This will cause temporary caller impedance, which can have a knock
		// on effect in processing.
		if !bypassWarning {
			log.Printf("Logger warning: %v\n", errJobsChannelIsFull)
		}
		jobsChannel <- newJob
	}
```
So if the channel is full, and you close the channel via `CloseLogger`, all the logs > 10000 will be waiting to append on the `jobsChannel`, which is now closed.